src/ORM/Table.php

```diff
@@ -2531,9 +2531,11 @@ public function __get($property)
         $association = $this->_associations->get($property);
         if (!$association) {
             throw new RuntimeException(sprintf(
-                'Table "%s" is not associated with "%s"',
+                'Undefined property `%s`. ' .
+                'You have tried to use the `%s` property on %s, but that association does not exist.',
```

Comment:
If you don't mind an outside comment, I would keep this simple for the user.  "You did not set an association between X and Y.  I think talking about properties is less clear since this is strictly dealing with associations.